shift_detector/checks/statistical_checks/categorical_statistical_check.py

```diff
@@ -17,14 +20,42 @@ def chi2_test(part1: pd.Series, part2: pd.Series):
 
 class CategoricalStatisticalCheck(SimpleStatisticalCheck):
 
+    def __init__(self, significance=0.01, sample_size=None, use_equal_dataset_sizes=False, sampling_seed=0,
+                 use_binning=False, use_embedding=False):
+        self.use_binning = use_binning
+        self.use_embedding = use_embedding
+        super().__init__(significance, sample_size, use_equal_dataset_sizes, sampling_seed)
+
     def check_name(self) -> str:
         return 'Categorical Statistical Check'
 
     def statistical_test_name(self) -> str:
         return 'Chi^2-Test with Log-Likelihood (G-Test)'
 
     def data_to_process(self, store):
-        df1, df2, columns = store[LowCardinalityPrecalculation()]
+        if not self.use_binning:
+            cat1, cat2, cat_col = store[LowCardinalityPrecalculation()]
+        else:
+            cat1, cat2 = store[ColumnType.categorical]
+            cat_col = store.column_names(ColumnType.categorical)
+        df1 = cat1
+        df2 = cat2
+        columns = set(cat_col)
+        if self.use_binning:
```

Comment:
please try to concat this with the upper if such that there is only one if self.use_binning to improve readability